modules.d/01systemd-networkd/module-setup.sh

```diff
@@ -57,14 +57,23 @@ install() {
         "$systemdsystemunitdir"/systemd-networkd-wait-online.service \
         "$systemdsystemunitdir"/systemd-networkd-wait-online@.service \
         "$systemdsystemunitdir"/systemd-network-generator.service \
-        ip
+        ip sed grep
+
+    inst_simple "$moddir"/networkd-wait-online-initrd.service \
+        "$systemdsystemunitdir"/networkd-wait-online-initrd.service
+
+    inst_simple "$moddir"/99-default.network \
+        "$systemdnetworkconfdir"/99-default.network
+
+    inst_hook cmdline 99 "$moddir"/networkd-config.sh
+    inst_hook initqueue/settled 99 "$moddir"/networkd-run.sh
 
     # Enable systemd type units
     for i in \
         systemd-networkd.service \
         systemd-networkd.socket \
         systemd-network-generator.service \
-        systemd-networkd-wait-online.service; do
+        networkd-wait-online-initrd.service; do
```

Comment:
Could this be achieved as a systemd unit drop-in instead, similar to e.g. https://github.com/dracut-ng/dracut-ng/blob/main/modules.d/01systemd-hostnamed/99-systemd-networkd-dracut.conf ? 

Just a thought.

Reply:
My way of thinking was that I want to avoid making the initqueue depend on the native networkd-wait-online service, just in case something else pulls in a service that also depends on it.
That could lead to the initqueue never running, cause there is no network, so it locks up forever.

Reply:
I'm also not sure if a drop-in can sensibly add to the Wanted-By section so it ends up in the right section.

Reply:
We need to consider both `hostonly` and `non-hostonly` configurations.

In non-hostonly the user might also have some drop-ins that dracut specifically copies over to inside of the generated initrd, which as I understand will have no impact now - see https://github.com/dracut-ng/dracut-ng/blob/main/modules.d/01systemd-networkd/module-setup.sh#L84 .

Reply:
It does work fine as a drop-in from the looks of it, so gonna do that.